example_copy_test.go

```diff
@@ -215,6 +216,37 @@ func ExampleCopy_remoteToRemote() {
 	// sha256:7cbb44b44e8ede5a89cf193db3f5f2fd019d89697e6b87e8ed2589e60649b0d1
 }
 
+func ExampleCopy_remoteToRemoteWithMount() {
+	reg, err := remote.NewRegistry(remoteHost)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	src, err := reg.Repository(ctx, "source")
+	if err != nil {
+		panic(err) // Handle error
+	}
+	dst, err := reg.Repository(ctx, "target")
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tagName := "latest"
+
+	opts := oras.CopyOptions{}
+	// Enable cross-repository blob mounting
+	opts.WithMount("source", dst.(registry.Mounter), nil)
```

Comment:
What will happen if other target instead of `dst` is passed to `WithMount`?

Reply:
Let's say the true destination target is `dst` but instead they pass in `dst2`.  The blobs would be mounted or copied to `dst2` but the manifest would be copied to `dst` and fail due to missing blobs.

Reply:
Can we avoid that situation in the first place with a better API design?